pkg/component/runtime/manager.go

```diff
@@ -95,7 +95,8 @@ type Manager struct {
 	waitMx    sync.RWMutex
 	waitReady map[string]waitForReady
 
-	mx           sync.RWMutex
+	updateMx     sync.Mutex
```

Comment:
Small nit: There's multiple mutexes in this struct now. Can we have some brief comments explaining what they're used or?

Reply:
+1, reading the uses of them it became more obvious but adding some comments to avoid misuse is a good idea.

Reply:
Added the requested comments to the mutexes.